version.go

```diff
@@ -37,17 +38,34 @@ func pkgVersionFromGit(gitdir string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	cmd = exec.Command("git", "describe", "--abbrev=0")
+
+	// Find the most recent tag (whether annotated or not)
+	cmd = exec.Command("git", "describe", "--abbrev=0", "--tags")
 	cmd.Dir = gitdir
 	// 1.0~rc1 < 1.0 < 1.0+b1, as per
 	// https://www.debian.org/doc/manuals/maint-guide/first.en.html#namever
 	lastTag := "0.0~"
 	if lastTagBytes, err := cmd.Output(); err == nil {
-		lastTag = strings.TrimSpace(string(lastTagBytes)) + "+"
+		lastTag = strings.TrimSpace(string(lastTagBytes))
+
+		// Compare with the most recent annotated tag
+		cmd = exec.Command("git", "describe", "--abbrev=0")
+		cmd.Dir = gitdir
+		if lastAnnotatedTagBytes, err := cmd.Output(); err == nil {
+			lastAnnotatedTag := strings.TrimSpace(string(lastAnnotatedTagBytes))
+			if lastTag != lastAnnotatedTag {
+				log.Printf("WARNING: Lightweight tag %q found, but the most recent annotated tag is %q\n", lastTag, lastAnnotatedTag)
```

Comment:
Can we make this warning actionable for the user, i.e. suggest what they should do in order to get this resolved?

Also, we should explain what the problem with lightweight tags is.

I’m thinking of something like “Lightweight tags (“git tag v1.0”) are problematic e.g. because they are not considered in “git describe”. Please suggest for upstream to use “git tag v1.0 -as” or similar in order to create an annotated tag instead.”